[python-updates] 2023/12/02 major updates (python312Packages, pythonRuntimeDepsCheckHook)

mweinelt opened the pull request:
This is a bulk update of python packages, that includes major version bumps.

### TODO
- [x] `pythonRuntimDepsCheckHook` (see #270457)
- [x] Recurse into `python312Packages`, stop recursing into `python310Packages`
- [x] Prune `SETUPTOOLS_SCM_PRETEND_VERSION`
- [x] Review and merge staging changes on the [project board](https://github.com/NixOS/nixpkgs/projects/39?query=is%3Aopen+sort%3Aupdated-desc)
- [x] Re-review changes and consider them for merge (ETA: 2023-12-02, 12:00+ UTC)
- [x] Run bulk updater (ETA: 2023-12-02, 21:00+ UTC)
- [x] Start evaluation

Once the evaluation has finished, jobs will start building and breakage can be reviewed over at https://hydra.nixos.org/jobset/nixpkgs/python-updates.

Additionally, a review report will appear at https://malob.github.io/nix-review-tools-reports.

### Known Issues:
- [x] meson-python on python312 depends on https://github.com/NixOS/nixpkgs/pull/268583
- [x] numpy 1.26.2 still uses distutils in its buildsystem, needs research which commits we need to backport
- [x] pytest-asyncio 0.22.0 breaks [bleak](https://gist.github.com/mweinelt/790dd633069cee7cda71e36924169df9), [aioesphomeapi](https://gist.github.com/mweinelt/fe629d072d1e947d3f69cd66fd920170) and a few other packages - reverted!

### Latest eval report

https://shells.darmstadt.ccc.de/~hexa/93.html

fabaff committed:
python311Packages.elasticsearch: add optional-dependencies

- clean-up inputs
- add pythonImportsCheck
- equalize content

fabaff committed:
python311Packages.netaddr: refactor

fabaff committed:
python311Packages.aiodns: refactor

fabaff committed:
python311Packages.fastjsonschema: 2.16.2 -> 2.18.1

Diff: https://github.com/horejsek/python-fastjsonschema/compare/v2.16.2...v2.18.1

fabaff committed:
python311Packages.rpds-py: 0.10.3 -> 0.10.6

fabaff committed:
python311Packages.prettytable: 3.8.0 -> 3.9.0

Diff: https://github.com/jazzband/prettytable/compare/refs/tags/3.8.0...3.9.0

Changelog: https://github.com/jazzband/prettytable/releases/tag/3.9.0

fabaff committed:
python311Packages.pyelftools: 0.29 -> 0.30

Diff: https://github.com/eliben/pyelftools/compare/v0.29...v0.30

Changelog: https://github.com/eliben/pyelftools/blob/v0.30/CHANGES

fabaff committed:
python311Packages.cbor2: remove maintainer

fabaff committed:
python311Packages.bleach: refactor

matthiasbeyer committed:
python3Packages.regress: init at 0.4.5

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>
Co-Authored-By: Victor Engmark <victor@engmark.name>

matthiasbeyer committed:
check-jsonschema: 0.25.0 -> 0.27.3

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>
Co-Authored-By: Victor Engmark <victor@engmark.name>

fabaff committed:
python311Packages.google-cloud-storage: refactor

anthonyroussel committed:
python3Packages.botocore: add passthru.optional-dependencies.crt

anthonyroussel committed:
python311Packages.boto3: 1.28.57 -> 1.33.6

https://github.com/boto/boto3/compare/1.28.57...1.33.6

anthonyroussel committed:
python3Packages.s3transfer: 0.7.0 -> 0.8.2

https://github.com/boto/s3transfer/compare/0.7.0...0.8.2

anthonyroussel committed:
awscli: 1.30.2 -> 1.31.6

https://github.com/aws/aws-cli/compare/1.30.2...1.31.6

anthonyroussel committed:
python311Packages.moto: update meta.homepage

anthonyroussel committed:
python3Packages.cfn-lint: enable test fixed upstream

anthonyroussel committed:
python3Packages.cfn-lint: update src.repo,meta.homepage

anthonyroussel committed:
python3Packages.aws-sam-translator: fix build for darwin

fabaff committed:
python311Packages.constantly: refactor

fabaff committed:
python311Packages.pyprusalink: refactor

fabaff committed:
python311Packages.amberelectric: refactor

SuperSandro2000 commented on pkgs/development/interpreters/python/hooks/python-runtime-deps-check-hook.py:
```diff
@@ -0,0 +1,97 @@
+    metadata = get_metadata(args.wheel)
+    tests = [test_requirement(requirement) for requirement in metadata.requires_dist]
+
+    if not all(tests):
```
We shoud list some of the possible options for people here like buildNpmPackage

SuperSandro2000 replied:
You got exactly what I wanted to say, no need to write romans. :)

Yes, exactly.

We could print something like 

```
Please try to use pythonRelaxDepsHook to release the version constraints if the library allows it and tests continue to pass.
If the hook does not work for some reason, please try to fall back to manually patching in postPatch.
```

mweinelt replied:
> You got exactly what I wanted to say, no need to write romans. :)

It took me more than a minute to grok though, and that happens more often than you think.

SuperSandro2000 replied:
hmmmm, I try to remember that in the future but that's just my normal writing style. I am not leaving things out on purpose.

SuperSandro2000 commented on pkgs/development/python-modules/setuptools-scm/setup-hook.sh:
```diff
@@ -0,0 +1,32 @@
+# Let built package know its version.
```
reposting here to not leave the comments in the closed PR:
Should we clean up SETUPTOOLS_SCM_PRETEND_VERSION now?

mweinelt replied:
Working on that, will push a treewide commit in a few minutes.

SuperSandro2000 commented on pkgs/development/python-modules/psycopg/default.nix:
```diff
@@ -50,14 +50,6 @@ let
       libc = "${stdenv.cc.libc}/lib/libc.so.6";
     })
-
-    (fetchpatch {
```
If I still could, I would just have pushed nit fixup to remove the unused input 🙃

mweinelt replied:
Fixed in 1327fd5accf1ae5ebee90a17023077fea352a8f5

dotlambda committed:
python311Packages.playwright: let setuptools-scm generate version file

mweinelt committed:
python3Packages.wcwidth: 0.2.6 -> 0.2.12

mweinelt committed:
python3Packages.wavefile: 1.5 -> 1.6.2

https://github.com/vokimon/python-wavefile#version-history

mweinelt committed:
python3Packages.websockets: 11.0.3 -> 12.0

https://github.com/aaugustin/websockets/blob/12.0/docs/project/changelog.rst

mweinelt committed:
python3Packages.websocket-client: 1.6.3 -> 1.6.4

https://github.com/websocket-client/websocket-client/blob/v1.6.4/ChangeLog

mweinelt committed:
python3Packages.wrapt: 1.14.1 -> 1.16.0

mweinelt committed:
python3Packages.wheel: 0.41.1 -> 0.42.0

mweinelt committed:
python3Packages.willow: 1.6.2 -> 1.7.0

mweinelt committed:
python3Packages.wsme: 0.11.0 -> 0.12.1

https://pythonhosted.org/WSME/changes.html

mweinelt committed:
python3Packages.wtforms: 3.0.1 -> 3.1.1

https://github.com/wtforms/wtforms/blob/3.1.1/CHANGES.rst

mweinelt committed:
python3Packages.xarray: 2023.8.0 -> 2023.11.0

mweinelt committed:
python3Packages.xxhash: 3.3.0 -> 3.4.1

mweinelt committed:
python3Packages.xformers: 0.0.22.post7 -> 0.03

https://github.com/facebookresearch/xformers/blob/0.03/CHANGELOG.md

mweinelt committed:
python3Packages.xhtml2pdf: 0.2.11 -> 0.2.13

https://github.com/xhtml2pdf/xhtml2pdf/releases/tag/v0.2.13

mweinelt committed:
python3Packages.yapf: 0.40.1 -> 0.40.2

https://github.com/google/yapf/blob/v0.40.2/CHANGELOG.md

mweinelt committed:
python3Packages.xyzservices: 2023.7.0 -> 2023.10.1

https://github.com/geopandas/xyzservices/releases/tag/2023.10.1

mweinelt committed:
python3Packages.zeroc-ice: 3.7.9.1 -> 3.7.10

mweinelt committed:
python3Packages.zfec: 1.5.7.2 -> 1.5.7.4

mweinelt committed:
python3Packages.zipp: 3.16.2 -> 3.17.0

mweinelt committed:
python3Packages.zope-i18nmessageid: 6.0.1 -> 6.1.0

https://github.com/zopefoundation/zope.i18nmessageid/blob/6.1.0/CHANGES.rst

mweinelt committed:
python3Packages.zope-hookable: 5.4 -> 6.0

mweinelt committed:
python3Packages.zstandard: 0.21.0 -> 0.22.0

mweinelt committed:
cc2538-bsl: 2023-08-14 -> 2.1-unstable-2023-10-03

fabaff committed:
python311Packages.fastnumbers: 5.0.1 -> 5.1.0

Diff: https://github.com/SethMMorton/fastnumbers/compare/refs/tags/5.0.1...5.1.0

Changelog: https://github.com/SethMMorton/fastnumbers/blob/5.1.0/CHANGELOG.md

mweinelt committed:
python312Packages.array-record: disable

We package a wheel, and there is none for 3.12.

anthonyroussel committed:
python3Packages.aiobotocore: fix build for darwin

mweinelt committed:
python312Packages.sphinxcontrib-jquery: fix tests

by replacing the previous 7.2 compat patch with another one that also
enables Python 3.12 compat.

mweinelt committed:
python311Packages.sphinx: disable racy tests

mweinelt committed:
python312Packages.eventlet: fix build and use pep517 builder

mweinelt committed:
python311Packages.gunicorn: use pep517 builder, expose extras

mweinelt committed:
python311Packages.pyzmq: fix build

mweinelt committed:
python312Packages.nosexcover: disable

mweinelt committed:
python311Packages.gitdb: relax smmap constraint

Checking runtime dependencies for gitdb-4.0.11-py3-none-any.whl
  - smmap<6,>=3.0.1 not satisfied by version 6.0.0

mweinelt committed:
python312Packages.future: apply patch for imp removal compat

mweinelt committed:
python312Packages.sure: disable tests

They use nose and nose requires the imp module.

mweinelt committed:
python311Packages.testscenarios: fix build

by providing pbr and switch to pep517 builder.

mweinelt committed:
python311Packages.pyroma: 3.2 -> 4.2

Refactor, convert to PEP 517 build, migrate to pytest and disable on
Python 3.12, because it relies on distutils.

https://github.com/regebro/pyroma/blob/4.2/CHANGES.txt

mweinelt committed:
python311Packages.tokenize-rt: 4.2.1 -> 5.2.0

https://github.com/asottile/tokenize-rt/compare/v4.2.1...v5.2.0

mweinelt committed:
python311Packages.constantly: 15.1.0 -> 23.10.4

mweinelt committed:
python312Packages.pyu2f: patch test compatibility

mweinelt committed:
python311Packages.asn1crypto: 1.5.1 -> 1.5.1-unstable-2023.11.03

Fast forward to a version with Python 3.12 compatibility.

mweinelt committed:
python311Packages.autocommand: fix build

by dropping bogus setup.py.

fabaff committed:
python311Packages.screenlogicpy: refactor

mweinelt committed:
python311Packages.subunit: propagate extras

Checking runtime dependencies for python_subunit-1.4.2-py3-none-any.whl
  - extras not installed

mweinelt committed:
python312Packages.parameterized: fix build

by patching in python 3.12 compatibility.

mweinelt committed:
python312Packages.bottle: disable failing tests

mweinelt committed:
python311Packages.rapidfuzz: relax cython constraint

mweinelt commented:
Report for the latest eval on my private hydra, if someone wants to dig in.

https://shells.darmstadt.ccc.de/~hexa/55.html

mweinelt committed:
python311Packages.dbus-python: move from dbus to dbus-python

mweinelt commented on pkgs/development/python-modules/dbus/default.nix:
```diff
@@ -1,46 +1,85 @@
+, dbus-glib
+
+# tests
+, pygobject3
```
unused

mweinelt committed:
python311Packages.pytest: clean up deps, expose testing extra

mweinelt committed:
python312Packages.numpy: apply patch to remove distutils usage

This fixes the build on Python 3.12.

mweinelt committed:
python311Packages.pytest-subtests: fix build

Checking runtime dependencies for pytest_subtests-0.11.0-py3-none-any.whl
  - attrs not installed

mweinelt committed:
python311Packages.pillow: drop pyroma test dependency

It is used for code quality, but the tests case is disabled and pyroma
is not compatible with Python 3.12.

mweinelt committed:
python312Packages.pandas: use cython 3

This is a hack to allow pandas 2.1.3 to build on python 3.12, which is
only supported from 2.2.0

Cython 2 still calls into distutils, Cython 3 does not.

mweinelt committed:
python311Packages.pythran: 0.13.1 -> 0.14.0

https://github.com/serge-sans-paille/pythran/compare/0.13.1...0.14.0

mweinelt committed:
python312Packages.pythran: backport python3.12 support

mweinelt committed:
python311Packages.structlog: disable failing test

mweinelt committed:
python312Packages.flasgger: backport 3.12 support

mweinelt committed:
python312Packages.pytest-regressions: disable failing tests

mweinelt committed:
python311Packages.fastapi: disable failing tests

mweinelt committed:
python311Packages.django_4: 4.2.7 -> 4.2.8

https://docs.djangoproject.com/en/4.2/releases/4.2.8/

mweinelt committed:
python311Packages.django-tables2: 2.7.0 -> 2.7.0

https://github.com/jieter/django-tables2/blob/v2.7.0/CHANGELOG.md

dotlambda commented:
@mweinelt Can you restore my playwright change?

mweinelt commented:
@dotlambda Sorry, apparently it went missing during the last rebase. Pulled 71a4231 from the reflog, PTAL.

mweinelt committed:
python311Packages.aiounittest: use pep517 build, disable on py312

natsukium committed:
python311Packages.pypytools: add a missing dependency

natsukium committed:
python311Packages.vispy: cleanup dependencies

natsukium committed:
python311Packages.jira: cleanup dependencies

natsukium committed:
python311Packages.inflect: disable a failing test

natsukium committed:
python311Packages.cheroot: fix build with setuptools-scm>=8

natsukium committed:
python311Packages.ruyaml: fix build with setuptools-scm>=8

natsukium committed:
python311Packages.setuptools-scm-git-archive: mark as broken

mweinelt commented:
Latest eval report: https://shells.darmstadt.ccc.de/~hexa/63.html

sqlalchemy was flaky on my hydra builders and I restarted it five times, it has now succeeded.

natsukium committed:
python311Packages.astropy-extension-helpers: add missing dependencies

mweinelt committed:
python311Packages.executing: 1.2.0 -> 2.0.1

https://github.com/alexmojaki/executing/compare/v1.2.0...v2.0.1

natsukium committed:
python311Packages.astropy-iers-data: init at 0.2023.12.04.00.30.20

introduced as a dependency of astropy

mweinelt committed:
mercurial: fix build with python3.12

mweinelt committed:
python312Packages.ddt: disable tests

Tests rely on aiounittest, which is not compatible with Python 3.12
right now.

mweinelt committed:
python312Packages.geopy: fix build

dotlambda committed:
python311Packages.trio-asyncio: modernize expression

mweinelt committed:
python311Packages.bimmer-connected: disable failing test

mweinelt committed:
python312Packages.async-generator: disable failing test

There is little hope for this package, but jupyterhub still depends on
it, while upstream wants to deprecate it.

mweinelt committed:
python312Packages.overrides: disable failing test

mweinelt committed:
python312Packages.python-json-logger: disable failing tests

mweinelt committed:
python312Packages.ipython-genutils: disable

> Vestigial IPython utilities: DO NOT USE

https://github.com/ipython/ipython_genutils

fabaff committed:
python311Packages.pyasn1: refactor

fabaff committed:
python311Packages.websocket-client: refactor

fabaff committed:
python311Packages.websockets: refactor

fabaff committed:
python311Packages.xxhash: refactor

fabaff committed:
python311Packages.iso8601: add changelog to meta

mweinelt commented:
Latest eval report: https://shells.darmstadt.ccc.de/~hexa/64.html

mweinelt committed:
python311Packages.icecream: use pep517 build, enable tests

and establish python 3.12 compatibilty.

mweinelt committed:
python311Packages.pyarrow: relax setuptools_scm constraint

and enable PEP517 build.

mweinelt committed:
maturinBuildHook: use dist dir relative to cargoRoot

With `cargoRoot` set to a subdirectory of the source, where the
Cargo.{lock,toml} are found, the final mv would previously fail, since
the build results appear relative to cargoRoot, not to the original
build directory.

mweinelt committed:
python312Packages.pendulum: init at 3.0.0b1

Adds the new pendulum beta early, since it comes with support for
Python 3.12, while keeping Python<3.12 on 2.x until 3.0 gets its final
release.

fabaff committed:
python311Packages.diskcache: 5.4.0 -> 5.6.3

Diff: https://github.com/grantjenks/python-diskcache/compare/v5.4.0...v5.6.3

fabaff committed:
python311Packages.voluptuous: refactor

- add changelog to meta
- add format
- disable on unsupported Python releases

fabaff committed:
python311Packages.responses: 0.23.3 -> 0.24.1

Diff: https://github.com/getsentry/responses/compare/refs/tags/0.23.3...0.24.1

Changelog: https://github.com/getsentry/responses/blob/0.24.1/CHANGES

fabaff committed:
python311Packages.requests-cache: 1.1.0 -> 1.1.1

Diff: https://github.com/requests-cache/requests-cache/compare/refs/tags/v1.1.0...v1.1.1

Changelog: https://github.com/requests-cache/requests-cache/blob/v1.1.1/HISTORY.md

fabaff committed:
python311Packages.freezegun: 1.2.2 -> 1.3.1

Changelog: https://github.com/spulec/freezegun/blob/1.3.1/CHANGELOG

fabaff committed:
python311Packages.httplib2: 0.21.0 -> 0.22.0

Diff: https://github.com/httplib2/httplib2/compare/v0.21.0...v0.22.0

natsukium committed:
python312Packages.joblib: fix build with python3.12

natsukium committed:
python311Packages.openai: fix build

natsukium committed:
python312Packages.pytest-openfiles: fix build with python312

astropy, which depended on this package, no longer needs it and
it will be archived in the near future

mweinelt commented:
@fabaff `elasticsearch` does not build like that, and it is at the same version as `elasticsearch8`, so having both packages is rather pointless.

natsukium committed:
python312Packages.pyquery: disable failing test

mweinelt commented:
Latest eval report: https://shells.darmstadt.ccc.de/~hexa/66.html

natsukium committed:
python311Packages.polars: fix build with maturinBuildHook changes

fabaff commented:
> @fabaff elasticsearch does not build like that, and it is at the same version as elasticsearch8, so having both packages is rather pointless.

As far as I remember was `elasticsearch8` introduced when `elasticsearch` was still at 7.x. Would make sense to remove one.

Sorry for the new build failure. Will fix [c899467](https://github.com/NixOS/nixpkgs/pull/271586/commits/c899467efe4b92b844ea4c6c2f613e311ddc97fa).

mweinelt commented:
Makes sense. But I think now that elasticsearch has been on 8.x we should only keep one and update all references. Can you take care of that?

fabaff committed:
python311Packages.python-docx: refactor

natsukium committed:
python312Packages.aspectlib: ignore deprecation warnings in tests

natsukium committed:
python312Packages.opt-einsum: fix build with python312

natsukium committed:
python312Packages.graphviz: fix tests with python3.12

natsukium committed:
python311Packages.pytest-subprocess: ignore deprecation warnings in tests

fabaff committed:
python311Packages.pontos: 23.12.0 -> 23.12.1

Diff: https://github.com/greenbone/pontos/compare/refs/tags/v23.12.0...v23.12.1

Changelog: https://github.com/greenbone/pontos/releases/tag/v23.12.1

natsukium committed:
python312Packages.js2py: disable

fabaff committed:
python311Packages.pytest-httpserver: 1.0.7 -> 1.0.8

Diff: https://github.com/csernazs/pytest-httpserver/compare/refs/tags/1.0.7...1.0.8

Changelog: https://github.com/csernazs/pytest-httpserver/blob/1.0.8/CHANGES.rst

fabaff committed:
python311Packages.pytest-httpx: refactor

fabaff committed:
python311Packages.pytest-cases: refactor

mweinelt commented:
Latest eval report: https://shells.darmstadt.ccc.de/~hexa/70.html

Also started linking them in the top post.

mweinelt committed:
python312Packages.jsonpickle: disable failing test

and refactor and convert to PEP517 build

mweinelt commented:
Did a rebase to resolve all the conflicts introduced by the PR that set explicit format on master. Also, Python 3.12.1 arrived in the meantime. Tried to move most conflicting packages to PEP517 builds, but I certainly failed here and there, we'll see.

Will trigger another eval now, but it will probably take a day until I can give feedback.

natsukium committed:
python311Packages.pytest-mock: disable failing test

mweinelt committed:
python311Packages.pytext-dist: disable flaky test

mweinelt committed:
python311Packagas.mutagen: provide setuptools for build-system

mweinelt committed:
python311Packages.pyjwt: provide setuptools

mweinelt committed:
pythonRelaxDepsHook: make distribution name matching case insensitive

Is it PyOpenSSL or pyOpenSSL or pyopenssl? With this change you don't
have to care. Write it as pYoPeNSsL for all I care¹.

Can we do it like that? Yes, we can. Distribution names should be matched
with `re.IGNORECASE` according to the wheel specification².

[1] https://packaging.python.org/en/latest/specifications/core-metadata/#name
[2] Just kidding, please make it look sane.

mweinelt committed:
python311Packages.aiocron: use pep517 builder

mweinelt committed:
python311Packages.curtsies: provide setuptools for build

mweinelt committed:
python311Packages.dynalite-devices: remove asynctest usage in testsuite

mweinelt committed:
python311Packages.lizard: disable failing tests

mweinelt committed:
python311Packages.mockfs: propgate importlib-metadata

Checking runtime dependencies for mockfs-1.1.4-py2.py3-none-any.whl
  - importlib-metadata not installed

mweinelt committed:
python311Packages.qmk-dotty-dict: 1.3.0.post1 -> 1.3.1

https://github.com/pawelzny/dotty_dict/releases/tag/v1.3.1

mweinelt committed:
python311Packages.wikitextparser: provide setuptools for build

mweinelt committed:
fixup! python3Packages.aenum: 3.1.12 -> 3.1.15

mweinelt committed:
python312Packages.ansiwrap: disable

Uses the imp module, which was removed in Python 3.12.

mweinelt committed:
python312Packages.antlr4-python3-runtime: fix tests

mweinelt committed:
python312Packages.py-ubjson: disable failing test

mweinelt committed:
python312Packages.bacpypes: disable

mweinelt committed:
python312Packages.bwrapy: disable

natsukium committed:
python311Packages.dbus-deviation: fix build

natsukium committed:
python312Packages.libcst: fix build

mweinelt committed:
hatch: fix build

mweinelt commented:
Disabled one more racy test in sphinx, which shadows many builds in https://shells.darmstadt.ccc.de/~hexa/77.html.

mweinelt committed:
python312Packages.screenlogicpy: disable blocking tests

dotlambda committed:
python311Packages.yarl: 1.9.3 -> 1.9.4

Changelog: https://github.com/aio-libs/yarl/blob/v1.9.4/CHANGES.rst

mweinelt committed:
python312Packages.amqtt: disable stuck tests

Hanging somewhere in epoll.

GaetanLepage committed:
python311Packages.transformers: 4.35.2 -> 4.36.1

Changelog: https://github.com/huggingface/transformers/releases/tag/v4.36.1

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/80.html

watchfiles has a test that times out after 10s, so it is a false-positive here.

GaetanLepage committed:
python311Packages.pylint: 3.0.2 -> 3.0.3

Changelog: https://github.com/pylint-dev/pylint/releases/tag/v3.0.3

mweinelt committed:
python312Packages.webtest: fix tests with python 3.12

mweinelt committed:
python312Packages.twisted: disable failing tests

mweinelt committed:
python311Packages.textfsm: fix package version

mweinelt committed:
python312Packages.mkdocs: fix build

mweinelt committed:
python312Packages.oscrypto: fix build

mweinelt committed:
python312Packages.nose: disable

Heavy reliance on the imp module, paired with being unmaintained.

mweinelt committed:
python312Packages.testresources: fix build

- Convert to PEP517 build
- Convert to unittestCheckHook
- Replace removed aliases in unittest module

mweinelt committed:
python311Packages.zeep: disable failing tests

mweinelt committed:
python311Packages.nose2: fix build

- Convert to PEP517 build
- Convert to unittestCheckHook
- Disable failing test
- Prune dependencies and expose coverage extra

mweinelt committed:
python311Packages.geojson: 3.0.1 -> 3.1.0

https://github.com/jazzband/geojson/blob/3.1.0/CHANGELOG.rst

mweinelt commented:
My hydra has finally caught up again. Latest eval includes everything up to and including the yarl 1.9.3 bump.

https://shells.darmstadt.ccc.de/~hexa/81.html

mweinelt committed:
python312Packages.scipy: test without nose

mweinelt committed:
python311Packages.docker: 6.1.3 -> 7.0.0

https://github.com/docker/docker-py/releases/tag/7.0.0

mweinelt committed:
python311Packages.moto: update bespoke test selection

mweinelt committed:
python312Packages.pyheos: disable tests stuck in epoll

mweinelt committed:
python311Packages.aioconsole: 0.6.2 -> 0.7.0

https://github.com/vxgmichel/aioconsole/releases/tag/v0.7.0

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/a-tale-of-getting-stuck/36731/2

mweinelt committed:
python311Packages.hatchling: 1.18.0 -> 1.19.0

https://github.com/pypa/hatch/releases/tag/hatchling-v1.19.0

mweinelt committed:
python311: 3.11.6 -> 3.11.7

https://docs.python.org/release/3.11.7/whatsnew/changelog.html

mweinelt committed:
python311Packages.pydevd: 2.9.6 -> 2.10.0

mweinelt committed:
python312Packages.pydevd: disable failing tests

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/85.html

The scipy failure was transient. The remaining issues are mostly valid.

mweinelt committed:
python311Packages.simplisafe-python: propagate certifi

Checking runtime dependencies for simplisafe_python-2023.10.0-py3-none-any.whl
  - certifi not installed

mweinelt committed:
python311Packages.pyairnow: prune build-system

The build happens using poetry-core, but setuptools and wheel were
previously required. They are fully redundant and I submitted a patch
upstream to remove them.

imincik committed:
python311Packages.pyqt5_sip: 12.11.0 -> 12.13.0

mweinelt committed:
python2Packages.wcwidth: fix eval

mweinelt committed:
python311Packages.rdflib: fix tests

mweinelt committed:
python312Packages.tqdm: disable

mweinelt committed:
borgbackup: pin msgpack at 1.0.4

Checking runtime dependencies for borgbackup-1.2.6-cp311-cp311-linux_x86_64.whl
  - msgpack!=1.0.1,<=1.0.5,>=0.5.6 not satisfied by version 1.0.7

mweinelt committed:
fasttext: 0.9.2 -> 0.9.2-unstable-2023-11-28

Fixes the build with GCC 13.

mweinelt committed:
python311Packages.hass-nabucas: relax acme constraint

Checking runtime dependencies for hass_nabucasa-0.74.0-py3-none-any.whl
  - acme==2.7.1 not satisfied by version 2.7.4

dotlambda committed:
python311Packages.pikepdf: 8.7.1 -> 8.9.0

Diff: https://github.com/pikepdf/pikepdf/compare/v8.7.1...v8.9.0

Changelog: https://github.com/pikepdf/pikepdf/blob/v8.9.0/docs/releasenotes/version8.rst

mweinelt committed:
python312Packages.dbus-python: disable

mweinelt committed:
python311Packages.jsonschema-spec: relax referencing constraint

mweinelt committed:
python311Packages.aiohappyeyeballs: init at 2.3.0

Happy Eyeballs for pre-resolved hosts

natsukium committed:
python312Packages.scikit-build: fix build with python3.12

mweinelt committed:
buku: use pep517 builder

mweinelt committed:
python311Packages.aioesphomeapi: regroup arguments

Please leave them as they are, because it makes it easier for me to check
up on them on updates.

Noticed while rebasing on python-updates...

mweinelt commented:
@dotlambda I reverted pytest-asyncio for now, 0.23.0 seems to have introduced quite some regressions.

mweinelt committed:
python311Packages.pytest-asyncio: prune args, add upgrade hint

pbsds committed:
python3Packages.gradio-client: 0.5.0 -> 0.7.3

pbsds committed:
python3Packages.gradio: 4.7.1 -> 4.9.1

pbsds committed:
intel-media-sdk: fix gcc13 build with patch

pbsds committed:
python3Packages.psygnal: add missing checkPhase input

mweinelt committed:
acpic: fix build

by using pep517 builder and relaxing pbr constraint

mweinelt committed:
aws-sam-cli: fix build, relax deps

pbsds committed:
python3Packages.zarr: propagate msgpack

pbsds committed:
python3Packages.numcodecs: add msgpack to checkPhase

Increases test coverage of functionality used downstream in zarr

mweinelt committed:
ablog: ignore sphinx 9.0 deprecation warning for now

mweinelt committed:
python311Packages.aws-adfs: relax configparser dependency

Checking runtime dependencies for aws_adfs-2.9.0-py3-none-any.whl
  - configparser==5.3.0 not satisfied by version 6.0.0

pbsds committed:
google-cloud-cpp: fix build with patch from upstream

The patch is merged in 2.16.0, but bumping is not
trivial and my machine takes 1h12m57s to build this.

mweinelt committed:
python311Packages.pypdf: disable failing test

The test is missing fpdf2, but including it is not possible because it
would create a dependency cycle.

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/93.html

mweinelt committed:
python311Packages.clickhouse-cli: add setuptools build dep

mweinelt committed:
python311Packages.clickgen: attd missing attrs dependency

pbsds committed:
python3Packages.tokenizers: 0.14.1 -> 0.15.0

Diff: https://github.com/huggingface/tokenizers/compare/v0.14.1...v0.15.0

pbsds committed:
python3Packages.gradio-pdf: init at 0.0.3

pbsds committed:
python3Packages.datasets: remove pyarrow<14.0.1 vulnerability fix

mweinelt committed:
coercer: 1.6 -> 2.4.3

Fixes the build and relaxes the impacket constraint.

mweinelt committed:
python311Packages.psycopg: 3.1.14 -> 3.1.15

https://github.com/psycopg/psycopg/blob/3.1.15/docs/news.rst#current-release

mweinelt committed:
python311Packages.django-js-asset: 2.1 -> 2.2

https://github.com/matthiask/django-js-asset/compare/refs/tags/2.1...2.2

mweinelt committed:
python311Packages.vallox-websocket-api: relax websockets constraint

Checking runtime dependencies for vallox_websocket_api-4.0.2-py3-none-any.whl
  - websockets<12.0,>=9.1 not satisfied by version 12.0

mweinelt committed:
python311Packages.pytautulli: fix yarl 1.9.4 compat

mweinelt committed:
python311Packages.jaraco-abode: fix compat with jacaro.functools

mweinelt committed:
python311Packages.blinkpy: relax setuptools constraint

mweinelt committed:
python311Packages.aioguardian: propagate certifi

Checking runtime dependencies for aioguardian-2023.11.0-py3-none-any.whl
  - certifi not installed

mweinelt committed:
python311Packages.securesystemslib: relax hatchling constraint

mweinelt committed:
python311Packages.pyopenuv: propagate certifi

Checking runtime dependencies for pyopenuv-2023.11.0-py3-none-any.whl
  - certifi not installed

mweinelt committed:
python311Packages.regenmaschine: propagate certifi

Checking runtime dependencies for regenmaschine-2023.11.0-py3-none-any.whl
  - certifi not installed

mweinelt committed:
python311Packages.aioridwell: propagate certifi

Checking runtime dependencies for aioridwell-2023.10.0-py3-none-any.whl
  - certifi not installed

mweinelt committed:
python311Packages.pyrmvtransport: disable failing test

mweinelt committed:
home-assistant: pin pyaussiebb, voluptuous, xbox-webapi

mweinelt committed:
xapian: enable parallel building

Cuts build time from 17 minutes down to 5 minutes on an Intel i7-8700k.

mweinelt commented:
Latest eval: https://shells.darmstadt.ccc.de/~hexa/97.html

Xapian does build on the two machines I have locally, just fails on the machines connected to my hydra instance.

mweinelt committed:
python311Packages.sigstore: init at 2.1.0

Packaged as a dependency for aiogithubapi.

mweinelt committed:
python311Packages.aiogithubapi: propagate sigstore

mweinelt committed:
python311Packages.omegaconf: add attrs to check deps

ImportError while loading conftest '/build/source/tests/conftest.py'.
tests/__init__.py:6: in <module>
    import attr
E   ModuleNotFoundError: No module named 'attr'

mweinelt committed:
python311Packages.toggl-cli: skip bulk updates

mweinelt committed:
python311Packages.sphinx-prompt: skip bulk update

mweinelt committed:
ansible-lint: relax ruamel.yaml constraint

mweinelt committed:
python311Packages.avidtools: drop typing dependency

This release includes a dependency on typing without a constraint on the
python version, but typing became obsolete in Python 3.5, so it does not
exist in nixpkgs.

mweinelt committed:
instaloader: add setuptools as native buile dep

mweinelt committed:
mitmproxy2swagger: relax ruamel.yamel constraint

mweinelt committed:
python311Packages.appthreat-vulnerability-db: relax msgpack constraint

afh commented:
❓ [babel 2.14.0](https://github.com/python-babel/babel/releases/tag/v2.14.0) was released on Dec 12. What would it take to include that release in this update?

mweinelt committed:
python311Packages.broadbean: disable racy test

afh commented:
Thank you, @mweinelt, for updating babel to 2.14.0 in this PR, much appreciated! 😃

mweinelt committed:
python311Packages.bbox: disable racy test

and some light reformatting.

mweinelt committed:
python311Packages.bytewax: 0.17.1 -> 0.17.2

https://github.com/bytewax/bytewax/releases/tag/v0.17.2

mweinelt committed:
python311Packages.asyncio-mqtt: skip bulk updates

Later versions were renamed and packaged under the aiomqtt name.

mweinelt committed:
python311Packges.ansible-vault-rw: fix build

mweinelt committed:
python311Packages.hatchling: 1.19.0 -> 1.21.0

https://github.com/pypa/hatch/releases/tag/hatchling-v1.21.0

natsukium committed:
python312Packages.propka: fix build with python3.12

natsukium committed:
python312Packages.propka: switch to pypa build

natsukium committed:
python312Packages.testfixtures: fix build with python3.12

natsukium committed:
python311Packages.testfixtures: switch to pypa build

mweinelt committed:
hatch: 1.7.0 -> 1.9.0

https://github.com/pypa/hatch/blob/hatch-v1.9.0/docs/history/hatch.md

trofi commented:
2884f6262ba5638254c6751bb8aa90cdacf34a78 added a few duplicate entries, caused evaluation failures like:

```
       error: duplicate formal function argument 'setuptools'

       at /mnt/archive/big/git/nixpkgs-staging-next/pkgs/development/python-modules/aioconsole/default.nix:8:3:

            7| , pythonOlder
            8| , setuptools
             |   ^
            9| }:
```

Proposed fix as:
- [] `python3Packages.aioconsole` https://github.com/NixOS/nixpkgs/pull/275797:

trofi commented:
Bisect says 536b7d10392b57a40f2a005f5dfc90e1744c25cf `python311Packages.pytest: clean up deps, expose testing extra` broke `libei` in `staging-next`:

```
$ nix build --no-link -f. -L libei
...
libei> Program python3 (attr, jinja2) found: NO modules: jinja2
libei> proto/meson.build:17:15: ERROR: python3 is missing modules: attr
libei> A full log can be found at /build/source/build/meson-logs/meson-log.txt
```

mweinelt commented:
On mobile right now, but a fix was referenced in the staging-next PR.

mattfawcett commented:
`pythonPackages.pyosmium` no longer builds with these upgrades due to it trying to patch something already patched in the latest version. Created https://github.com/NixOS/nixpkgs/pull/281389 to fix it.

jonringer commented:
> @mweinelt , I found [this commit](https://github.com/NixOS/nixpkgs/commit/ee7f05659540612cdcdbf3dc88219ea48df960e3) merged as a part of this PR updating python3Packages.rasterio to version 4 which [doesn't exist](https://github.com/rasterio/rasterio/releases). Package with this change doesn't even build ! It looks very suspicious to me !

This should have only been included if they didn't mark that tag as a `pre-release`. Looks like it was already reverted but the forced update of all packages will sometimes get these errors propagating to us.

jonringer commented:
although, looked at that branch... and it's from 9 years ago, not sure why it did get included.... the release metadata should be getting fetched from github.

mweinelt commented:
We arrive at the following using https://api.github.com/repos/rasterio/rasterio/releases and because we determine the highest version in `_determine_latest_version`.

```
print(_get_latest_version_github("python311Packages.rasterio", "rasterio", "", "1.3.8", "major"))
('4', '07j1fba1jm55qby0ljxmm983n229qgarjdm9gwyfzdq4cag0bvb0', 'release-test-')
```

CertainLach commented:
xformers upgrade is broken here:
https://github.com/NixOS/nixpkgs/pull/271586/commits/5dde5f31da583f36e7f79d0b9177b9faa8662f0a

Version 0.03 exists, but this is a typo in upstream repository:
![image](https://github.com/NixOS/nixpkgs/assets/6235312/a49b7d5a-d861-47a6-ac52-f71f19f72d4a)

And in fact, xformers version is downgraded here
0.0.22.post7 => 0.03

Version 0.0.22.post7 is much newer:
![image](https://github.com/NixOS/nixpkgs/assets/6235312/482004c4-5951-41de-8b16-fa567718f3ce)
